ui/tests/e2e/extensions.spec.js

```diff
@@ -0,0 +1,249 @@
+    );
+    console.log('Successfully navigated to Meshery Docker Extension');
+    await newPage.waitForLoadState('networkidle');
+    await newPage.close();
+    console.log('Download Button is working Fine');
+  });
+
+  test('Verify Meshery Desing Embed Details', async ({ page }) => {
+    // Verify the Leanr More Button works or not
+    const learnMoreButton = page.locator('button.MuiButtonBase-root:has-text("Learn More")');
+    await expect(learnMoreButton).toBeVisible();
+    console.log('Learn More button is visible');
+
+    await expect(learnMoreButton).toBeEnabled();
+    console.log('Learn More button is enabled');
+
+    await Promise.all([page.waitForEvent('popup'), learnMoreButton.click()]);
+    const newPage = await page.context().pages()[1];
+
+    // Verify URL
+    await expect(newPage).toHaveURL('https://docs.layer5.io/kanvas/designer/embedding-designs/');
+    console.log('Successfully navigated to Meshery Design Embed');
+    await newPage.waitForLoadState('networkidle');
+    await newPage.close();
+    console.log('Learn More Button is working Fine');
+  });
+
+  test('Verify Meshery Catalog Section Details', async ({ page }) => {
+    // First verify the toggle button functionality
+    const toggleButton = page
+      .locator('div:has-text("Meshery Catalog") >> input[type="checkbox"]')
+      .nth(0);
+    // console.log(await toggleButton.count());
+    await page.waitForLoadState('networkidle');
+    const istoggleVisible = await toggleButton.isVisible();
+
+    if (istoggleVisible) {
+      console.log('Toggle button is visible');
+
+      // Check whether the Toggle Button is checked or Not BEFORE Clicking it
+      const isCheckedBefore = await toggleButton.isChecked();
+      console.log(`Toggle button Status: ${isCheckedBefore ? 'Enable' : 'Disable'}`);
+
+      // Click the Button
+      await toggleButton.click();
+      console.log('Toggle button clicked');
+
+      // Check whether the Toggle Button is checked or Not AFTER Clicking it
+      const isCheckedAfter = await toggleButton.isChecked();
+      console.log(`Toggle button Status: ${isCheckedAfter ? 'Enable' : 'Disable'}`);
+    } else {
+      console.log('Toggle button is not visible');
+      throw new Error('Toggle button not found');
+    }
+
+    // Verify the Meshert Catalog Link
+    const catalogLink = page.locator('a[href="https://meshery.io/catalog"]');
+
+    // Verify the Link is visible
+    await expect(catalogLink).toBeVisible();
+    console.log('Meshery Catalog Link is visible');
+
+    // Verify the link text
+    const linkText = await page.locator('text=Explore the Meshery Catalog').isVisible();
+    expect(linkText).toBeTruthy();
+    console.log('Link text is correct');
+
+    // Click the link and verify navigation
+    await Promise.all([
+      // Wait for the new page/tab to open
+      page.waitForEvent('popup'),
+      // Click the link
+      catalogLink.click(),
+    ]);
```

Comment:
curious if this promise.all fails will it give us a good error message. 

might be worth separating if not